model.py

```diff
@@ -6937,6 +6925,26 @@ def catalog_identifier(self, _db, identifier):
             self.catalog.append(identifier)
             _db.commit()
 
+    def works_updated(self, _db):
+        """Returns all of a collection's works that have been updated since the
+        last time the collection was checked"""
+        if self.catalog:
+            catalog_identifier_ids = [identifier.id for identifier in self.catalog]
+
+            # Find the works in the catalog that have been created or updated
+            # since the collection last checked.
+            query = _db.query(Work).join(Work.coverage_records)
+            query = query.join(Work.license_pools).join(Identifier)
+            query = query.filter(Identifier.id.in_(catalog_identifier_ids))
```

Comment:
There are going to be hundreds of thousands of Identifiers in a collection, right? Rather than iterate over all of them collecting their IDs, you should join Identifier against collections_identifiers against Collection and let the database do the work.